### DISCUSSION TITLE
Is there a command line arg or an env var to make "brew install" and "brew upgrade" output summary of errors in the end?

### DISCUSSION THREAD

**Discussion by yermulnik** [Homebrew/discussions#4900] [Everyday usage]
### Output of `brew config`

```shell
not related
```

### Output of `brew doctor`

```shell
not related
```

### Description of issue

When one uses `brew install` or `brew upgrade` to install/upgrade a number of packages, there might be a case where one or more packages encountered errors e.g. at post-install stage (like errors with linking because of already existing targets). Though the output goes beyond the visual part of the terminal and hence user cannot notice there were errors thrown and hence user misses to fix them (or at least to catch them).
Is there an option to collect and output a summary of install/upgrade/reinstall/uninstall/etc failures at the very end of command run, so that user is notified explicitly and can do whatever they think is needed?
Obviously user can scroll terminal history all the way up, though doing this every time user runs simple `brew upgrade` on a number of packages is tedious.
Thanks for the pointers.

**Comment by SMillerDev** [comment]
I believe errors are printed to stderr, so you could capture that

**Reply by yermulnik** [reply]
Thanks for the pointer. Although this would require some sort of thin wrapper that copies stderr of `brew` command to a var or a file and outputs its content at the end of `brew` run if the var or file is not empty. What I was thinking of was to have such capability bundled into `brew`.
Hope one day this will be implemented if there's a demand for that from more people than just me 😺
